app/controllers/CyaMaintainProfileController.scala

```diff
@@ -74,6 +74,35 @@ class CyaMaintainProfileController @Inject() (
       case Left(errors) =>
         Future.successful(logErrorsAndContinue(errorMessage, continueUrl, errors))
     }
+  }
+
+  def onPageLoadNiphls(): Action[AnyContent] = (identify andThen getData andThen requireData) { implicit request =>
+    TraderProfile.validateNiphlsUpdate(request.userAnswers) match {
+      case Right(_)     =>
+        val list = SummaryListViewModel(
+          rows = Seq(
+            HasNiphlSummary.rowUpdate(request.userAnswers),
+            NiphlNumberSummary.rowUpdate(request.userAnswers)
+          ).flatten
+        )
+        Ok(view(list, routes.CyaMaintainProfileController.onSubmitNiphls))
+      case Left(errors) =>
+        logErrorsAndContinue(errorMessage, continueUrl, errors)
+    }
+  }
 
+  def onSubmitNiphls(): Action[AnyContent] = (identify andThen getData andThen requireData).async { implicit request =>
+    TraderProfile.validateNiphlsUpdate(request.userAnswers) match {
+      case Right(niphlNumber) =>
+        traderProfileConnector.getTraderProfile(request.eori).flatMap { traderProfile =>
+          val updatedProfile = traderProfile.copy(niphlNumber = niphlNumber)
```

Comment:
if we use build niphls we don't need to build the trader profile here because the function already does this

Reply:
It seems to end up complicating things rather than simplifying them

Reply:
Fair enough - could you also remove the build niphls and tests if it no longer used?

Reply:
Good point, removed